uber/templates/regform.html

```diff
@@ -114,41 +114,47 @@
     </div>
 {% endif %}
 
-{% if c.DONATIONS_ENABLED and c.PREREG_DONATION_OPTS and c.PAGE_PATH != '/registration/form' %}
+{% if c.DONATIONS_ENABLED and c.PAGE_PATH != '/registration/form' %}
     <script type="text/javascript">
         var donationChanged = function () {
-            setVisible('.affiliate-row', $.val('amount_extra') > 0);
+            updateAmountExtra();
+            setVisible('.affiliate-row', $.val('donation_tier') != {{ c.NO_DONATION }});
             {% if attendee.gets_shirt %}
                 setVisible('.shirt-row', true);
             {% else %}
-                setVisible('.shirt-row', $.val('amount_extra') >= {{ c.SHIRT_LEVEL }});
+                setVisible('.shirt-row', $.inArray($.val('donation_tier'), {{ c.SHIRT_TIERS }}) != -1);
```

Comment:
These should stay as-is, in cases where an attendee with a free donation level is visiting their confirmation page.

Come to think of it, we need to accommodate for that when updating amount_extra too.